libstorage.apib

```diff
@@ -1010,34 +1107,59 @@ Internal server error
 
     + Schema
 
-            { "$ref": "https://raw.githubusercontent.com/emccode/libstorage/master/libstorage.json#/definitions/internalServerError" }
+            { "$ref": "https://raw.githubusercontent.com/codedellemc/libstorage/master/libstorage.json#/definitions/internalServerError" }
 
-## Volume Inspector [/volumes/{service}/{volumeID}]
+## Volume Inspector [/volumes/{service}/{volumeID}?{attachments}]
 A single Volume object. A central part of the libStorage
 API, a Volume resource represents a storage volume.
 
 + Parameters
 
-    + service: `ec2-00` (string, required)
+    + service: `ebs-00` (string, required)
 
         The name of the service to which the Volume belongs
 
     + volumeID: `vol-000` (string, required)
 
         The volume's unique ID
 
+    + attachments (string,optional)
+
+        A value that indicates how and if volume attachments should be retrieved.
+        <br/><br/>
+        The `attachments` parameter can be any of the following values:
+        <br/>
+        Value | Description
+        ------|------------
+        0 | No attachment information is requested. This is the default value and the same as omitting this parameter altogether.
+        1 | Attachment information is requested. This value indicates attachment information should be provided for all volumes.
+        2 | _My_ attachment information is requested. This value indicates attachment information should be provided for volumes attached to the instance specified in the [instance ID](#def-instance-id) request header should be returned. If this bit is set then the instance ID header is required.
+        4 | Attempt to map devices provided via the [local devices](#def-local-devices) request header to the appropriate attachment information. If this bit is set then the instance ID and local device headers are required.
```

Comment:
Will it be up to the driver to know when these values are appropriate? Values `2` and `4` could be weird in this scenario -- if the client is requesting info about a volume that isn't attached to itself, but you add flags that try to map or otherwise require info from the client.